tests/integration/ha_tests/test_self_healing.py

```diff
@@ -46,14 +47,20 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
     """Build and deploy three unit of PostgreSQL."""
     # It is possible for users to provide their own cluster for HA testing. Hence, check if there
     # is a pre-existing cluster.
-    if await app_name(ops_test):
-        return
-
-    charm = await ops_test.build_charm(".")
-    async with ops_test.fast_forward():
-        await ops_test.model.deploy(charm, resources={"patroni": "patroni.tar.gz"}, num_units=3)
-        await ops_test.juju("attach-resource", APP_NAME, "patroni=patroni.tar.gz")
-        await ops_test.model.wait_for_idle(status="active", timeout=1000)
+    if not await app_name(ops_test):
+        charm = await ops_test.build_charm(".")
+        async with ops_test.fast_forward():
+            await ops_test.model.deploy(
+                charm, resources={"patroni": "patroni.tar.gz"}, num_units=3
+            )
+            await ops_test.juju("attach-resource", APP_NAME, "patroni=patroni.tar.gz")
+            await ops_test.model.wait_for_idle(status="active", timeout=1000)
+    # Deploy the continuous writes application charm if it wasn't already deployed.
+    if await app_name(ops_test, APPLICATION_NAME) is None:
+        async with ops_test.fast_forward():
+            charm = await ops_test.build_charm("tests/integration/ha_tests/application-charm")
+            await ops_test.model.deploy(charm, application_name=APPLICATION_NAME)
+            await ops_test.model.wait_for_idle(status="active", timeout=1000)
```

Comment:
Moved this out of the fixture, so that the suite will fail from the get go if the app charm fails to build. When in the fixture, tests will continue to execute even if the suite cannot be successful.

Reply:
Nice! Just as a suggestion: having only one call to `await ops_test.model.wait_for_idle(status="active", timeout=1000)` at the end outside both `if` blocks may speedup this part of the tests.